### DISCUSSION TITLE
ipython-sql sql jupyter extension is not compatible with sqlachemy v2

### DISCUSSION THREAD

**Discussion by centeiol** [sqlalchemy/sqlalchemy#8676] [Usage Questions]
### Describe the bug

Hello I get the error hen connecting with oracle+oracledb 

__init__() got an unexpected keyword argument 'bind'
Connection info needed in SQLAlchemy format, example:
               postgresql://username:password@hostname/dbname
               or an existing connection: dict_keys([])

### To Reproduce

```python
import oracledb
import pandas as pd
import sqlalchemy




%reload_ext sql
#%sql oracle+oracledb://user:pass@host-scan.domain.net:1521/?service_name=XPTO.domain.net

sqlalchemy.__version__
```

### Error

__init__() got an unexpected keyword argument 'bind'
Connection info needed in SQLAlchemy format, example:
               postgresql://username:password@hostname/dbname
               or an existing connection: dict_keys([])
'2.0.0b1'

### Versions

- OS: Windows 10 PRO
- Python: Python 3.9.1
- SQLAlchemy:'2.0.0b1'
- Database:Oracle 12c 
- DBAPI (eg: psycopg, cx_oracle, mysqlclient):oracle+oracledb

**Comment by CaselIT** [comment]
Hi,

There is no stack trace and no reproducing code provided, so we cannot offer any help. Please provide a reproduction script and/or a complete stack trace

**Comment by centeiol** [comment]
Hello , 
I am just a Dba trying to put ( notebook with  Visual Studio ) working to graph some data.
Before I used cx_oracle without problem , but after install of python-oracledb and version '2.0.0b1' Of Alchemy , nothing is working.
For security reasons user, pass , host-scan.domain.net and XPTO.domain.net are not the real ones.



I try to dump a stack like this 
```py
import oracledb
import pandas as pd
import sqlalchemy
import traceback



%reload_ext sql
%sql oracle+oracledb://user:pass@host-scan.domain.net:1521/?service_name=XPTO.domain.net
#traceback.print_exc()
traceback.extract_stack()

Result:

__init__() got an unexpected keyword argument 'bind'
Connection info needed in SQLAlchemy format, example:
               postgresql://username:password@hostname/dbname
               or an existing connection: dict_keys([])
[<FrameSummary file c:\app\python\lib\runpy.py, line 197 in _run_module_as_main>,
 <FrameSummary file c:\app\python\lib\runpy.py, line 87 in _run_code>,
 <FrameSummary file C:\Users\luicen\AppData\Roaming\Python\Python39\site-packages\ipykernel_launcher.py, line 17 in <module>>,
 <FrameSummary file c:\app\python\lib\site-packages\traitlets\config\application.py, line 982 in launch_instance>,
 <FrameSummary file C:\Users\luicen\AppData\Roaming\Python\Python39\site-packages\ipykernel\kernelapp.py, line 712 in start>,
 <FrameSummary file c:\app\python\lib\site-packages\tornado\platform\asyncio.py, line 215 in start>,
 <FrameSummary file c:\app\python\lib\asyncio\base_events.py, line 596 in run_forever>,
 <FrameSummary file c:\app\python\lib\asyncio\base_events.py, line 1890 in _run_once>,
 <FrameSummary file c:\app\python\lib\asyncio\events.py, line 80 in _run>,
 <FrameSummary file C:\Users\luicen\AppData\Roaming\Python\Python39\site-packages\ipykernel\kernelbase.py, line 510 in dispatch_queue>,
 <FrameSummary file C:\Users\luicen\AppData\Roaming\Python\Python39\site-packages\ipykernel\kernelbase.py, line 499 in process_one>,
 <FrameSummary file C:\Users\luicen\AppData\Roaming\Python\Python39\site-packages\ipykernel\kernelbase.py, line 406 in dispatch_shell>,
 <FrameSummary file C:\Users\luicen\AppData\Roaming\Python\Python39\site-packages\ipykernel\kernelbase.py, line 730 in execute_request>,
 <FrameSummary file C:\Users\luicen\AppData\Roaming\Python\Python39\site-packages\ipykernel\ipkernel.py, line 383 in do_execute>,
 <FrameSummary file C:\Users\luicen\AppData\Roaming\Python\Python39\site-packages\ipykernel\zmqshell.py, line 528 in run_cell>,
 <FrameSummary file c:\app\python\lib\site-packages\IPython\core\interactiveshell.py, line 2885 in run_cell>,
 <FrameSummary file c:\app\python\lib\site-packages\IPython\core\interactiveshell.py, line 2940 in _run_cell>,
 <FrameSummary file c:\app\python\lib\site-packages\IPython\core\async_helpers.py, line 129 in _pseudo_sync_runner>,
 <FrameSummary file c:\app\python\lib\site-packages\IPython\core\interactiveshell.py, line 3139 in run_cell_async>,
 <FrameSummary file c:\app\python\lib\site-packages\IPython\core\interactiveshell.py, line 3318 in run_ast_nodes>,
 <FrameSummary file c:\app\python\lib\site-packages\IPython\core\interactiveshell.py, line 3378 in run_code>,
 <FrameSummary file C:\Users\luicen\AppData\Local\Temp\ipykernel_22172\2233386577.py, line 13 in <module>>]

```

Regards

**Reply by zzzeek** [reply]
hi -

I'm guessing you are trying to run SQLAlchemy 2.0 and you have code like this somehwere:

``MetaData(bind=engine)``

while your stack trace is incomplete and does not show the actual line of code, that's usually what the error you see means.

Before upgrading to SQLAlchemy 2.0 you need to go through the [migration guide](https://docs.sqlalchemy.org/en/20/changelog/migration_20.html), the specific issue that this error looks like would be: https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#implicit-and-connectionless-execution-bound-metadata-removed

**Reply by CaselIT** [reply]
this seems some extension or something of jupyter notebook that allows using magic commands to connect using slqlachemy.
I have never seen it, so you will probably need to ask for support from them.

The issue here is likely sqlalchemy 2 not oracle db. You should see the same error if you try to connect using cx_oracle.

**Reply by CaselIT** [reply]
This seems to be the extension https://github.com/catherinedevlin/ipython-sql

And mike is right as rain: https://github.com/catherinedevlin/ipython-sql/blob/24f13561f4527a931f9f0c43aed93ff1b40cba78/src/sql/connection.py#L52

@centeiol I suggest raising the issue to them

**Reply by centeiol** [reply]
Hello all, 

Sorry for my late awnser 
Thank you very much, I will open a new thread at ipython-sql.

Regards

**Comment by centeiol** [comment]
Hi 

Thanks
Actualy is SQLAlchemy   SQLAlchemy:'2.0.0b1' 
That I installed via 
pip install --pre SQLAlchemy

I will look at the link.

Regards
